src/bin/lockin-internal-demo.rs

```diff
@@ -0,0 +1,294 @@
+#![deny(warnings)]
+#![no_std]
+#![no_main]
+#![cfg_attr(feature = "nightly", feature(core_intrinsics))]
+
+use stm32h7xx_hal as hal;
+
+#[macro_use]
+extern crate log;
+
+use rtic::cyccnt::{Instant, U32Ext};
+
+use heapless::{consts::*, String};
+
+// A constant sinusoid to send on the DAC output.
+const DAC_SEQUENCE: [f32; 8] =
+    [0.0, 0.707, 1.0, 0.707, 0.0, -0.707, -1.0, -0.707];
+
+use dsp::{iir, iir_int, lockin::Lockin, reciprocal_pll::TimestampHandler};
+use hardware::{Adc1Input, Dac0Output, Dac1Output, AFE0, AFE1};
+use stabilizer::{
+    hardware, server, ADC_SAMPLE_TICKS_LOG2, SAMPLE_BUFFER_SIZE_LOG2,
+};
+
+const SCALE: f32 = ((1 << 15) - 1) as f32;
+
+const TCP_RX_BUFFER_SIZE: usize = 8192;
+const TCP_TX_BUFFER_SIZE: usize = 8192;
+
+#[rtic::app(device = stm32h7xx_hal::stm32, peripherals = true, monotonic = rtic::cyccnt::CYCCNT)]
+const APP: () = {
+    struct Resources {
+        afes: (AFE0, AFE1),
+        adc1: Adc1Input,
+        dacs: (Dac0Output, Dac1Output),
+        net_interface: hardware::Ethernet,
+
+        #[init([0.; 5])]
+        iir_state: iir::IIRState,
+
+        #[init(iir::IIR { ba: [1., 0., 0., 0., 0.], y_offset: 0., y_min: -SCALE - 1., y_max: SCALE })]
+        iir: iir::IIR,
+
+        pll: TimestampHandler,
+        lockin: Lockin,
+    }
+
+    #[init]
+    fn init(c: init::Context) -> init::LateResources {
+        // Configure the microcontroller
+        let (mut stabilizer, _pounder) = hardware::setup(c.core, c.device);
+
+        let pll = TimestampHandler::new(
+            4, // relative PLL frequency bandwidth: 2**-4, TODO: expose
+            3, // relative PLL phase bandwidth: 2**-3, TODO: expose
+            ADC_SAMPLE_TICKS_LOG2 as usize,
+            SAMPLE_BUFFER_SIZE_LOG2,
+        );
+
+        let lockin = Lockin::new(
+            &iir_int::IIRState::default(), // TODO: lowpass, expose
```

Comment:
```suggestion
            &iir_int::IIRState([21101, 42202, 21101, 2137942692, -1064243070]),
```

is a 1e-3 corner frequency butterworth lowpass

Reply:
Would you mind providing context as to how the values were generated so I can add a reference in the code?

Reply:
This is `lowpass_iir_coefficients(1e-3, 1/2f64.sqrt(), 2.)` from `dsp/src/lockin.rs`. We can reasonably move that into `dsp/src/iir_int.rs` and port it to `cossin()`. It doesn't neet do be fast.

Reply:
https://github.com/quartiq/stabilizer/blob/c769bdbd4cb5759fff601b2a99e6822a1a31c421/src/bin/lockin.rs#L59